### PR TITLE
`MapDomain`: refactor tests to `@testing-library/react`

### PR DESCRIPTION
**Pull request by flootr**
#### Proposed Changes

* `MapDomain`: refactor tests to `@testing-library/react`

#### Testing Instructions

* Verify unit tests still pass

Related to #63409

### CODE REVIEW

File: client/my-sites/domains/map-domain/test/map-domain.js [hunk 1]
```diff
@@ -31,65 +43,82 @@ describe( 'MapDomain component', () => {
...
+		await userEvent.click( backBtn );
 		expect( pageSpy ).toBeCalledWith( '/domains/add/baba' );
 	} );
 
 	test( 'does not render a notice by default', () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } /> );
-		// we match the notice by props, because enzyme isn't matching the Notice type for some reason
-		expect( wrapper.find( { status: 'is-error' } ) ).toHaveLength( 0 );
+		render( <MapDomain { ...defaultProps } /> );
+		expect( screen.queryByLabelText( 'Notice' ) ).toBeNull();
```
Comment (tyxla) [inline client/my-sites/domains/map-domain/test/map-domain.js]: Perhaps we can be a bit more specific about what we're asserting here?

```suggestion
		expect( screen.queryByLabelText( 'Notice' ) ).not.toBeInTheDocument();
```

File: client/my-sites/domains/map-domain/test/map-domain.js [hunk 2]
```diff
@@ -31,65 +43,82 @@ describe( 'MapDomain component', () => {
...
+		expect( screen.queryByTestId( 'map-domain-step' ) ).toBeInTheDocument();
 	} );
 
-	test( "goes back when HeaderCake's onClick is fired", () => {
-		const wrapper = shallow( <MapDomain { ...defaultProps } /> );
-		expect( wrapper.find( HeaderCake ).prop( 'onClick' ) ).toEqual( wrapper.instance().goBack );
+	test( "goes back when HeaderCake's onClick is fired", async () => {
+		render( <MapDomain { ...defaultProps } /> );
+		// @TODO be more specific maybe to omit the `getALLByRole`
+		const [ backBtn ] = screen.getAllByRole( 'button', { name: /back/i } );
```
Comment (tyxla) [inline client/my-sites/domains/map-domain/test/map-domain.js:70]: I think we can totally go with `getByRole()` - there's a single button there. Same for the queries below.

**Review by tyxla — APPROVED** [review]
LGTM 👍  Just a couple of optional suggestions.

File: client/my-sites/domains/map-domain/test/map-domain.js (hunk 2 above)
Comment (flootr) [reply client/my-sites/domains/map-domain/test/map-domain.js:70]: There is a second button rendered by one of the child components. We could mock it out, but I guess it'd make that test less integrated. Also, I saw I forgot to tick my own TODO :D

Comment (tyxla) [reply client/my-sites/domains/map-domain/test/map-domain.js:70]: Haha, well, you could mock it out indeed. Sounds like an odd reason to mock it in the first place, but I don't see why not.

**Pull request merged**